roles/matrix-nginx-proxy/tasks/validate_config.yml

```diff
@@ -18,3 +18,9 @@
     msg: >-
       `matrix_ssl_retrieval_method` needs to be set to a known value.
   when: "matrix_ssl_retrieval_method not in ['lets-encrypt', 'self-signed', 'manually-managed', 'none']"
+
+- name: Fail on unknown matrix_nginx_proxy_ssl_config
+  fail:
+    msg: >-
+      `matrix_nginx_proxy_ssl_config` needs to be set to a known value.
+  when: "matrix_nginx_proxy_ssl_config not in ['Custom', 'Old', 'Intermediate', 'Modern']"
```

Comment:
Besides this check, we should probably add another one that verifies if `matrix_nginx_proxy_ssl_custom_protocols` is set when `matrix_nginx_proxy_ssl_preset == 'custom'`.